Don't let "Edit Report" button overlap error message

NoahCarnahan opened the pull request:
Diff is a bit messy because I also rearranged some logic, but the only real changes are this:
- Added div with clearfix around (floated) button
- Added margin-bottom to button

cc @orangejenny

NoahCarnahan committed:
Don't let edit report button overlap error message

NoahCarnahan commented:
Looked like this before:
<img width="1142" alt="screen shot 2015-08-24 at 10 26 50 am" src="https://cloud.githubusercontent.com/assets/2117127/9443233/f53eb1d2-4a4d-11e5-800a-89d46c5b046f.png">

now looks like this:
<img width="1147" alt="screen shot 2015-08-24 at 10 26 13 am" src="https://cloud.githubusercontent.com/assets/2117127/9443239/fd007842-4a4d-11e5-8952-c38b5ca4c540.png">

NoahCarnahan commented on corehq/apps/userreports/templates/userreports/configurable_report.html:
```diff
@@ -98,18 +98,20 @@
-            <a class="btn btn-primary pull-right" href="
-                {% url 'edit_report_in_builder' domain report.report_config_id %}
-            ">{% trans "Edit Report" %}</a>
+            <a class="btn btn-primary" style="margin-bottom: 8px" href="
```
I would have preferred to use some site-wide css class for adding this margin (I chose 8px arbitrarily), but there isn't such a class that I know of. Is there a best practice for this sort of thing?

orangejenny commented:
Ohhh, I get it now (screenshots, so helpful). Page-level warnings / info / errors are meant to be above page controls & content, therefore `.alert` has a bottom margin, but not a top margin. So this is really a case of awkward styling indicating that there's a problem with the page's markup: the right thing to do is to move the edit button below the error.

I didn't look closely at the templates, but maybe dividing up main_column in base_template_new.html - say, a block for alerts, and then one for page content, with sub-templates overriding the new content block instead of main_column - is the way to go.

NoahCarnahan committed:
Move report alerts above Edit Report button

NoahCarnahan commented:
Ah good point, thanks! Moved button below alerts; removed inline style.

<img width="1143" alt="screen shot 2015-08-24 at 1 51 19 pm" src="https://cloud.githubusercontent.com/assets/2117127/9447841/715a79ea-4a67-11e5-9ecd-a530516d0e2c.png">

orangejenny commented:
Pedantic, but `built-warning` would fit better in the alerts block rather than the main content block.

Looks good once tests pass.

benrudolph commented on corehq/apps/reports_core/templates/reports_core/base_template_new.html:
```diff
@@ -79,7 +79,6 @@
                     }
                 },
                 error: function(jqXHR, textStatus, errorThrown) {
-                    console.log("wah waaaah");
```
lol

NoahCarnahan committed:
move built-warning to appropriate template block

NoahCarnahan commented:
Agreed. Warning moved.
